backend/src/apiserver/template/v2_template.go

```diff
@@ -217,3 +255,16 @@ func (t *V2Spec) RunWorkflow(modelRun *model.Run, options RunWorkflowOptions) (u
 	executionSpec.SetPodMetadataLabels(util.LabelKeyWorkflowRunId, options.RunId)
 	return executionSpec, nil
 }
+
+func isKubernetesExecutorConfig(template []byte) bool {
+	var platformSpec pipelinespec.PlatformSpec
+	templateJson, err := yaml.YAMLToJSON(template)
+	if err != nil {
+		return false
+	}
+	err = protojson.Unmarshal(templateJson, &platformSpec)
```

Comment:
I believe `yaml.YAMLToJSON(template)` will take the first part of the yaml file. So I expect this function always fail for input like:
```
<YAML1>
---
<YAML2>
```

This is probably expected, but we might need to add comment explaining this behavior.

Reply:
This function is only called with individual yaml docs, because the input is already split in `NewV2SpecTemplate()`. I'm not sure if we still need to add comment in this case.

Reply:
That makes sense. My comment was a nit.